### DISCUSSION TITLE
Option. Learning rates per technology

### DISCUSSION THREAD

**Discussion by pz-max** [PyPSA/technology-data#49] [Ideas]
Would it make sense to add a script that can handle learning rates of different shape per technology?

For instance, let's assume I find a dataset for 2030 including costs for new technologies. Now, I want to run PyPSA for 2050 which means at the moment that the cost will be adjusted to 2050 by an overall identical inflation rate (2\%). In the best case, the user would like to have the option to create costs for each technology based on a "learning by learning factor" and a "learning by deployment factor" (i.e. [Kittner & Kammen 2017, Nature paper](https://www.nature.com/articles/nenergy2017125?proof=t%252525C2%252525A0)).

The need for such an option might be not there so much for established technologies since many publications share long-term data for that purpose, however, when exploring novel technologies this might desirable function :)

Happy to contribute this in early summer, since I would need this!

![image](https://user-images.githubusercontent.com/61968949/154142666-328996d3-211a-4710-8174-ddd91180afa9.png)

**Comment by euronion** [comment]
We're currently mainly using the DEA data with some additional data from various sources. All of these sources already make assumptions on learning (if not progressive, then at least the "no learning" assumption). Adding such a functionality might increase the risk of misuse, e.g. applying learning rates twice to the same technology.

What you can do:
* use the `manual_input.csv` (https://github.com/PyPSA/technology-data/blob/master/inputs/manual_input.csv)  to overwrite existing entries and provide own assumptions, e.g. based on your desired learning curve.

(Not an opinion, just a remark ;)

**Reply by pz-max** [reply]
@euronion, I understand the concerns. In the case of the DEA data, a learning adjustment makes no sense. Further, to increase stability and consistency, the DEA data should be rather untouched or just updated with their reports.

Context:
My plan is to add a large data collection of energy storage data to the repository. Many of these technologies are not tracked by DEA. I think it would be a sustainable contribution if one could add such technology data to this repository. Thereby, I agree one should never interfere with the standard DEA data.

The plan:
- Provide function/multi-function with a collection of new storage technology data (often single year estimations). Maybe in a subfolder inputs/energy-storage
- Add a function that converts the single year estimation of multiple-new technologies (not available at DEA) with a learning rate to multi-year assumptions
- Create an option that adds the above data to the DEA data
- Make adjustments in the config.yaml
- Add documentation

**Reply by lisazeyen** [reply]
Hi Max, great to hear that your are planning to add a collection of new storage technology data. With a simple learning rate you don't receive new cost estimates, since they do depend on how much of the storage capacity is going to be build. At the end you are going to rely on some sort of forecast, just as the DEA assumptions. If you have some assumptions about how the installed capacity is going to be in the upcoming years you could add a function where you can modify the learning rate and calculate the resulting investment costs.
